modules/autoscaler/kube-resoures.tf

```diff
@@ -0,0 +1,260 @@
+  }
+
+  rule {
+    api_groups = ["coordination.k8s.io"]
+    resources  = ["leases"]
+    verbs      = ["create"]
+  }
+
+  rule {
+    api_groups     = ["coordination.k8s.io"]
+    resource_names = ["cluster-autoscaler"]
+    resources      = ["leases"]
+    verbs          = ["get", "update"]
+  }
+}
+
+resource "kubernetes_role" "cluster-autoscaler" {
+  metadata {
+    name      = "cluster-autoscaler"
+    namespace = "kube-system"
+    labels = {
+      "k8s-addon" = "cluster-autoscaler.addons.k8s.io"
+      "k8s-app"   = "cluster-autoscaler"
+    }
+  }
+
+  rule {
+    api_groups = [""]
+    resources  = ["configmaps"]
+    verbs      = ["create", "list", "watch"]
+  }
+
+  rule {
+    api_groups     = [""]
+    resources      = ["configmaps"]
+    resource_names = ["cluster-autoscaler-status", "cluster-autoscaler-priority-expander"]
+    verbs          = ["delete", "get", "update", "watch"]
+  }
+}
+
+resource "kubernetes_cluster_role_binding" "cluster-autoscaler" {
+  metadata {
+    name = "cluster-autoscaler"
+    labels = {
+      "k8s-addon" = "cluster-autoscaler.addons.k8s.io"
+      "k8s-app"   = "cluster-autoscaler"
+    }
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = "cluster-autoscaler"
+  }
+  subject {
+    kind      = "ServiceAccount"
+    name      = "cluster-autoscaler"
+    namespace = "kube-system"
+  }
+}
+
+resource "kubernetes_role_binding" "cluster-autoscaler" {
+  metadata {
+    name      = "cluster-autoscaler"
+    namespace = "kube-system"
+    labels = {
+      "k8s-addon" = "cluster-autoscaler.addons.k8s.io"
+      "k8s-app"   = "cluster-autoscaler"
+    }
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "Role"
+    name      = "cluster-autoscaler"
+  }
+  subject {
+    kind      = "ServiceAccount"
+    name      = "cluster-autoscaler"
+    namespace = "kube-system"
+  }
+}
+
+resource "kubernetes_deployment" "cluster-autoscaler" {
+  metadata {
+    name      = "cluster-autoscaler"
+    namespace = "kube-system"
+    labels = {
+      app = "cluster-autoscaler"
+    }
+
+    annotations = {
+      "cluster-autoscaler.kubernetes.io/safe-to-evict" = "false"
+    }
+  }
+
+  spec {
+    replicas = 1
+
+    selector {
+      match_labels = {
+        app = "cluster-autoscaler"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = "cluster-autoscaler"
+        }
+        annotations = {
+          "prometheus.io/scrape" = "true"
+          "prometheus.io/port"   = "8085"
+        }
+      }
+
+      spec {
+        priority_class_name = "system-cluster-critical"
+        security_context {
+          run_as_non_root = "true"
+          run_as_user     = "65534"
+          fs_group        = "65534"
+        }
+        service_account_name = "cluster-autoscaler"
+        container {
+          image = "k8s.gcr.io/autoscaling/cluster-autoscaler:v1.23.0"
```

Comment:
have this formed based on cluster version and patch passed from external by default 0